tests/system/test_system.py

```diff
@@ -814,6 +871,33 @@ def test_download_w_generation_match(self):
 
         self.assertEqual(file_contents, stored_contents)
 
+    def test_download_w_failed_crc32c_checksum(self):
+        blob = self.bucket.blob("FailedChecksumBlob")
+        file_contents = b"Hello World"
+        blob.upload_from_string(file_contents)
+        self.case_blobs_to_delete.append(blob)
+
+        temp_filename = tempfile.mktemp()
```

Comment:
I just opened #247 to track removing the unsafe `tempfile.mktemp`.  Here, you would use `tempfile.NamedTemporaryFile`. e.g.:

```python
        with tempfile.NamedTemporaryFile() as temp_f:

            # Intercept the digest processing at the last stage and replace it with garbage
            with mock.patch(
                "google.resumable_media._helpers.prepare_checksum_digest",
                return_value="FFFFFF==",
            ):
                with self.assertRaises(resumable_media.DataCorruption):
                    blob.download_to_filename(temp_f.name, checksum="crc32c")
```

etc.